components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
-    logical :: state_debug_checks
-
-    logical :: l_tracer_aero
-    logical :: l_vdiff
-    logical :: l_rayleigh
-    logical :: l_gw_drag
-    logical :: l_ac_energy_chk
-
-    !
-    !-----------------------------------------------------------------------
-    !
-    lchnk = state%lchnk
-    ncol  = state%ncol
-
-    nstep = get_nstep()
-    
-    call phys_getopts( do_clubb_sgs_out       = do_clubb_sgs, &
-                       state_debug_checks_out = state_debug_checks &
-                      ,l_tracer_aero_out      = l_tracer_aero      &
-                      ,l_vdiff_out            = l_vdiff            &
-                      ,l_rayleigh_out         = l_rayleigh         &
-                      ,l_gw_drag_out          = l_gw_drag          &
-                      ,l_ac_energy_chk_out    = l_ac_energy_chk    &
-                     )
-
-    ! Adjust the surface fluxes to reduce instabilities in near sfc layer
-    if (phys_do_flux_avg()) then 
-       call flux_avg_run(state, cam_in,  pbuf, nstep, ztodt)
-    endif
-
-    ! Validate the physics state.
-    if (state_debug_checks) &
-         call physics_state_check(state, name="before tphysac")
-
-    call t_startf('tphysac_init')
-    ! Associate pointers with physics buffer fields
-    itim_old = pbuf_old_tim_idx()
-
-
-    ifld = pbuf_get_index('DTCORE')
-    call pbuf_get_field(pbuf, ifld, dtcore, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
-
-    call pbuf_get_field(pbuf, tini_idx, tini)
-    call pbuf_get_field(pbuf, qini_idx, qini)
-    call pbuf_get_field(pbuf, cldliqini_idx, cldliqini)
-    call pbuf_get_field(pbuf, cldiceini_idx, cldiceini)
-
-    ifld = pbuf_get_index('CLD')
-    call pbuf_get_field(pbuf, ifld, cld, start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
-
-    ifld = pbuf_get_index('AST')
-    call pbuf_get_field(pbuf, ifld, ast, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
-
-    !
-    ! accumulate fluxes into net flux array for spectral dycores
-    ! jrm Include latent heat of fusion for snow
-    !
-#ifdef MAML
-    !do the average of cam_in surface fluxes over num_inst_atm land instances
-    factor_xy = 1._r8 / dble(num_inst_atm)
-    shfavg_in = 0._r8
-    lhfavg_in = 0._r8
-    wsxavg_in = 0._r8
-    wsyavg_in = 0._r8
-    do i=1,ncol
-       do ii=1,num_inst_atm
-          tend%flx_net(i) = tend%flx_net(i) + (( cam_in%shf(i,ii) +               &
-             ((cam_out%precc(i,ii)  + cam_out%precl(i,ii) ) * latvap*rhoh2o ) +      &
-             ((cam_out%precsc(i,ii) + cam_out%precsl(i,ii)) * latice*rhoh2o )) * factor_xy)
-          shfavg_in(i) = shfavg_in(i) + cam_in%shf(i,ii)*factor_xy
-          lhfavg_in(i) = lhfavg_in(i) + cam_in%lhf(i,ii)*factor_xy
-          wsxavg_in(i) = wsxavg_in(i) + cam_in%wsx(i,ii)*factor_xy
-          wsyavg_in(i) = wsyavg_in(i) + cam_in%wsy(i,ii)*factor_xy
-       end do
-    end do
-#else
-    do i=1,ncol
-       tend%flx_net(i) = tend%flx_net(i) + cam_in%shf(i) + (cam_out%precc(i) &
```

Comment:
It looks like we lost this computation of the net flux, was that intentional?

Reply:
Correct, this appears to be part of the "code graveyard" that's a leftover from the original spectral dycro used in CAM, and is no longer used anywhere. There's actually one extra call that I meant to remove, so I'll do that now.